bouncer/util.py

```diff
@@ -70,4 +101,35 @@ def parse_document(document):
             _("The annotation has an invalid document URI"),
             "uri_not_a_string")
 
-    return (annotation_id, document_uri)
+    return {
+            "annotation_id": annotation_id,
+            "document_uri": document_uri,
+            "can_reveal_metadata": can_reveal_metadata,
+            "quote": quote,
+            "text": text
+            }
+
+
+def make_pretty_url(url):
+    """
+    Return the domain name from `url` for display.
+    """
+    try:
+        parsed_url = parse.urlparse(url)
+        pretty_url = parsed_url.netloc[:NETLOC_MAX_LENGTH]
+        if len(parsed_url.netloc) > NETLOC_MAX_LENGTH:
+            pretty_url = pretty_url + jinja2.Markup("&hellip;")
+    except:
+        return "CannotParseURL"
```

Comment:
I think I wrote a comment saying that we should have error handling for this because parsing a URL can theoretically fail. I can't find that comment anymore, but there are two comments I have regarding this error handling:

1. We should avoid catching all errors (`except:`) as much as possible, this could easily mask a bug in the code that we will never notice. The [documentation for urllib.parse.urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) mentions that it raises `ValueError` in certain circumstances. So we should only check for `ValueError` and let any other exception blow up.

2. Returning the cryptic string "CannotParseURL" would, for example, show the string "Hypothesis annotation for CannotParseURL" to the user, which isn't really helpful. I think it would make more sense for this function to return `None` in case of a `ValueError` and then we show the normal boilerplate ("Hypothesis annotation") string in the view instead of the more detailed one.